File: bindings/csharp/src/ProvenPop3.cs
// SPDX-License-Identifier: PMPL-1.0-or-later
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// POP3 protocol types for proven-servers.

namespace Proven;

/// <summary>Command matching the Idris2 ABI tags (0-10).</summary>
public enum Command : byte
{
    User = 0,
    Pass = 1,
    Stat = 2,
    List = 3,
    Retr = 4,
    Dele = 5,
    Noop = 6,
    Rset = 7,
    Quit = 8,
    Top = 9,
    Uidl = 10
}

/// <summary>State matching the Idris2 ABI tags (0-2).</summary>
public enum State : byte
{
    Authorization = 0,
    Transaction = 1,
    Update = 2
}

/// <summary>Response matching the Idris2 ABI tags (0-1).</summary>
public enum Response : byte
{
    Ok = 0,
    Err = 1
}

/// <summary>Pop3Error matching the Idris2 ABI tags (0-5).</summary>
public enum Pop3Error : byte
{
    Ok = 0,
    InvalidSlot = 1,
    NotActive = 2,
    InvalidTransition = 3,
    InvalidCommand = 4,
    AuthFailed = 5
}

File: bindings/php/src/ProvenPop3.php
<?php
// SPDX-License-Identifier: MPL-2.0
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// POP3 protocol types for proven-servers.

declare(strict_types=1);

namespace ProvenServers;

/** Command matching the Idris2 ABI tags. */
enum Command: int
{
    case User = 0;
    case Pass = 1;
    case Stat = 2;
    case List = 3;
    case Retr = 4;
    case Dele = 5;
    case Noop = 6;
    case Rset = 7;
    case Quit = 8;
    case Top = 9;
    case Uidl = 10;
}

/** State matching the Idris2 ABI tags. */
enum State: int
{
    case Authorization = 0;
    case Transaction = 1;
    case Update = 2;
}

/** Response matching the Idris2 ABI tags. */
enum Response: int
{
    case Response_Ok = 0;
    case Err = 1;
}

/** Pop3Error matching the Idris2 ABI tags. */
enum Pop3Error: int
{
    case Pop3Error_Ok = 0;
    case InvalidSlot = 1;
    case NotActive = 2;
    case InvalidTransition = 3;
    case InvalidCommand = 4;
    case AuthFailed = 5;
}
